### PR TITLE
docs: Phase2 - convert providers sample to Standalone

### PR DESCRIPTION
**Pull request by wardbell**
The `providers.md` page that goes with the `providers` sample code is listed in NgModules section. It seems to be the NgModule version of `dependency-injection-providers.md`.

I didn’t realize that this guide was in the NgModules section until AFTER I had converted the sample code to Standalone and modified the text of `providers.md`.

In so doing, I improved the sample to show how to provide services in a route setup. This is actually working code, unlike the markdown version in `guide/standalone-components.md` !!!

Overall, I think the text and code changes are an improvement, properly inform the NgModules reader, and should be incorporated in the updated docs.

However, should you disagree, you can easily discard these changes as this commit is in its own PR.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by dylhunn — APPROVED** [review]
Not sure if I'm the right reviewer for this. Perhaps Emma or Mark is?

File: aio/content/guide/providers.md [hunk 1]
```diff
@@ -18,85 +18,65 @@ ng generate service User
...
+<code-example header="src/app/users.component.ts" path="providers/src/app/users.component.ts" region="component-providers"></code-example>
 
-<img alt="any-provider-scope" class="left" src="generated/images/guide/providers/any-provider.svg">
+Angular destroys the service when the providing component is destroyed.
 
-</div>
+This can be useful when want to interact with a service in multiple components simultaneously and each component's service instance must have its own internal state.
 
-## Limiting provider scope with components
+## Providing services in NgMmodules
```
Comment (bougwal) [inline aio/content/guide/providers.md:72]: nit: typo
```suggestion
## Providing services in NgModules
```

**Review by bougwal — APPROVED** [review]

File: aio/content/guide/providers.md (hunk 1 above)
Comment (wardbell) [reply aio/content/guide/providers.md]: Good catch. Fixed.

File: aio/content/guide/providers.md
Comment (MarkTechson) [inline aio/content/guide/providers.md:1]: I see that you've had to make some updates to reference new files that were created. Can we keep the changes to the text minimal here? The original text does not appear to be in conflict with being migrated to standalone. We can add a contextual label, such as "For NgModule based apps..." and make a concise suggestion. Is that possible here?

**Review by MarkTechson — CHANGES_REQUESTED** [review]

**Pull request closed without merging**